concent_api/core/tests/test_utils.py

```diff
@@ -0,0 +1,107 @@
+import datetime
+import mock
+
+from django.conf import settings
+from django.test import override_settings
+from django.test import TestCase
+
+from golem_messages import constants
+from golem_messages import helpers
+
+from core.tests.utils import ConcentIntegrationTestCase
+from core.utils import calculate_maximum_download_time
+from core.utils import calculate_subtask_verification_time
+from utils.helpers import get_current_utc_timestamp
+from utils.helpers import parse_timestamp_to_utc_datetime
+from utils.testing_helpers import generate_ecc_key_pair
+
+(CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY) = generate_ecc_key_pair()
+
+
+class CalculateMaximumDownloadTimeTestCase(TestCase):
+
+    @override_settings(
+        CUSTOM_PROTOCOL_TIMES=False
+    )  # pylint: disable=no-self-use
+    def test_that_calculate_maximum_download_time_will_call_maximum_download_time_when_custom_protocol_times_is_off(self):
+        with mock.patch('core.utils.maximum_download_time', return_value=datetime.timedelta()) as mock_maximum_download_time:
+            calculate_maximum_download_time(size=1)
+
+        mock_maximum_download_time.assert_called_once_with(
+            1,
+            settings.MINIMUM_UPLOAD_RATE,
+        )
+
+    @override_settings(
+        CUSTOM_PROTOCOL_TIMES=True
+    )  # pylint: disable=no-self-use
+    def test_that_calculate_maximum_download_time_will_call_maximum_download_time_when_custom_protocol_times_is_on(self):
+        with mock.patch('core.utils.maximum_download_time', return_value=datetime.timedelta()) as mock_maximum_download_time:
+            calculate_maximum_download_time(size=1)
+
+        mock_maximum_download_time.assert_not_called()
+
+    @override_settings(
+        DOWNLOAD_LEADIN_TIME=int(constants.DOWNLOAD_LEADIN_TIME.total_seconds())
+    )
+    def test_that_both_maximum_download_time_implementation_should_return_same_result_when_golem_messages_constants_match_concent_settings(self):
```

Comment:
I would add some more "not so round" values, e. 7, 19, 331, etc.

Reply:
:+1: